src/ConnectedMode/ConnectionInfo.cs

```diff
@@ -28,5 +28,5 @@ public enum ServerType
         SonarCloud
     }
 
-    public record Connection(string id, ServerType serverType);
+    public record Connection(string id, ServerType serverType, bool hasSmartNotifications);
```

Comment:
On the caller side of this record
```
new Connection("https://sonarcloud.io/myOrg", ServerType.SonarCloud, false)
```
we pass obviously boolean values 😅 

However, I noticed that it was hard to understand what this `true` or `false` represents without looking at the record's definition. For code readability, it would be nice to replace it with an enum `SmartNotifications.Enabled`.

For comparison:
```
new Connection("https://sonarcloud.io/myOrg", ServerType.SonarCloud, SmartNotifications.Enabled)
```

Reply:
I understand your point, but I think it is not worth it to introduce an enum because it brings more complexity: we need to convert it to bool for the UI and then again convert it when passing it to SlCore.